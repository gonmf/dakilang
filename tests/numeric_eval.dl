test1(X, Res) :- eval(X, "($0 - 2) + 43 * (3 + $0)", Res).
test1(3, Res)?

test2(X, Res) :- eval(X, "-1-2-3-4-5-6-7+$0", Res).
test2(3, Res)?

test3(X, Res) :- eval(X, "(((5))) + (((-5))) + $0", Res).
test3(3, Res)?

test4(X, Res) :- eval(X, "3.14 * $0", Res).
test4(3, Res)?

test5(X, Res) :- eval(X, "$0 ^ ($0)", Res).
test5(3, Res)?

test6(X, Res) :- eval(X, "$0 / (6 - ($0 * 2))", Res).
test6(3, Res)?

test7(X, Res) :- eval(X, "$0 * 00", Res).
test7(3, Res)?

test8(X, Res) :- eval(X, "$0 + ~1 - 4", Res).
test8(3, Res)?

test9(X, Res) :- eval(X, "$0 ~1 - 3 * 5", Res).
test9(3, Res)?

test10(X, Res) :- eval(X, "~$0", Res).
test10(3, Res)?

test11(X, Res) :- eval(X, "~ 3 + $0", Res).
test11(3, Res)?

