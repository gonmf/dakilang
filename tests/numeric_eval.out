> test1(X, Res) :- eval(X, "($ - 2) + 43 * (3 + $)", Res).
> test1(3, Res)?
Res = 264

>
> test2(X, Res) :- eval(X, "-1-2-3-4-5-6-7+$", Res).
> test2(3, Res)?
Res = -25

>
> test3(X, Res) :- eval(X, "(((5))) + (((-5))) + $", Res).
> test3(3, Res)?
Res = 3

>
> test4(X, Res) :- eval(X, "3.14 * $", Res).
> test4(3, Res)?
Res = 9.42

>
> test5(X, Res) :- eval(X, "$ ^ ($)", Res).
> test5(3, Res)?
Res = 0

>
> test6(X, Res) :- eval(X, "$ / (6 - ($ * 2))", Res).
> test6(3, Res)?
No

>
> test7(X, Res) :- eval(X, "$ * 00", Res).
> test7(3, Res)?
Res = 0

>
> test8(X, Res) :- eval(X, "$ + ~1 - 4", Res).
> test8(3, Res)?
No

>
> test9(X, Res) :- eval(X, "$ ~1 - 3 * 5", Res).
> test9(3, Res)?
No

>
> test10(X, Res) :- eval(X, "~$", Res).
> test10(3, Res)?
Res = -4

>
> test11(X, Res) :- eval(X, "~ 3 + $", Res).
> test11(3, Res)?
Res = -1

>
